net/Socket.hpp

```diff
@@ -938,13 +938,24 @@ class SocketPoll
             SocketThreadOwnerChange::resetThreadOwner(*newSocket);
 
             std::lock_guard<std::mutex> lock(_mutex);
-            const bool wasEmpty = _newSockets.empty() && _newCallbacks.empty();
+            const bool wasEmpty = taskQueuesEmpty();
             _newSockets.emplace_back(std::move(newSocket));
             if (wasEmpty)
                 wakeup();
         }
     }
 
+    void transferSocketTo(const std::weak_ptr<Socket>& socket,
+                          const std::weak_ptr<SocketPoll>& toPoll,
+                          SocketDisposition::MoveFunction cb)
+    {
+        std::lock_guard<std::mutex> lock(_mutex);
+        const bool wasEmpty = taskQueuesEmpty();
+        _pendingTransfers.emplace_back(socket, toPoll, std::move(cb));
+        if (wasEmpty)
+            wakeup();
+    }
+
     /// Takes socket from @fromPoll and moves it to this current
     /// poll. Blocks until the transfer is complete.
     void takeSocket(const std::shared_ptr<SocketPoll> &fromPoll,
```

Comment:
Love it - of course we could and perhaps should use transferSocketTo for 'takeSocket' as well ? =) but that can be a whole separate patch-set.

Reply:
I have something like this to hand, but didn't want to clutter this pr up further. There's some extra complexity there in that the dest SocketPoll polling loop in that use case isn't actively running so takeSocket is waiting on notification of removal from the source SocketPoll rather than notification of addition to the dest SocketPoll.